src/Symfony/Component/Console/Command/Command.php

```diff
@@ -250,12 +253,22 @@ public function run(InputInterface $input, OutputInterface $output)
 
         $input->validate();
 
-        if ($this->code) {
-            $statusCode = call_user_func($this->code, $input, $output);
-        } else {
-            $statusCode = $this->execute($input, $output);
+        try {
+            if (!$this->lock($output)) {
+                $statusCode = 0;
+            } elseif ($this->code) {
+                $statusCode = call_user_func($this->code, $input, $output);
+            } else {
+                $statusCode = $this->execute($input, $output);
+            }
+        } catch (\Exception $e) {
+            $this->release();
+
+            throw $e;
         }
 
+        $this->release();
```

Comment:
if HHVM does not support ``finally``, it should not advocate itself as a PHP 5.6 version. So this should be discussed with the HHVM team.

Symfony 3 already relies on finally anyway, so I think HHVM supports it